From: olh@suse.de
Subject: another broken floppy 
References: SUSE49049

No recognition of USB-FDD

  <4>Linux version 2.6.8-24.10-default (geeko@buildhost) (gcc version 3.3.4 (pre 3.3.5 20040809)) #1 Wed Dec 22 11:54:27 UTC 2004
  <6>usb 1-1: Product: USB Floppy Drive
  <6>usb 1-1: Manufacturer: Sony
  <6>scsi1 : SCSI emulation for USB Mass Storage devices
  <5>  Vendor: SONY      Model: USB-FDU           Rev: 5.01
  <5>  Type:   Direct-Access                      ANSI SCSI revision: 02
  <6>scsi: unknown device type 31
  <5>  Vendor: SONY      Model: USB-FDU           Rev: 5.01
  <5>  Type:   Unknown                            ANSI SCSI revision: 02
  <5>Attached scsi removable disk sda at scsi1, channel 0, id 0, lun 0
  <5>Attached scsi generic sg0 at scsi1, channel 0, id 0, lun 0,  type 0
  <5>Attached scsi generic sg1 at scsi1, channel 0, id 0, lun 1,  type 31
  <6>scsi: unknown device type 31
  <5>  Vendor: SONY      Model: USB-FDU           Rev: 5.01
  <5>  Type:   Unknown                            ANSI SCSI revision: 02
  <5>Attached scsi generic sg2 at scsi1, channel 0, id 0, lun 2,  type 31
  <6>scsi: unknown device type 31
  <5>  Vendor: SONY      Model: USB-FDU           Rev: 5.01
  <5>  Type:   Unknown                            ANSI SCSI revision: 02
  <5>Attached scsi generic sg3 at scsi1, channel 0, id 0, lun 3,  type 31
  <6>scsi: unknown device type 31
  <5>  Vendor: SONY      Model: USB-FDU           Rev: 5.01
  <5>  Type:   Unknown                            ANSI SCSI revision: 02
  <5>Attached scsi generic sg4 at scsi1, channel 0, id 0, lun 4,  type 31
  <6>scsi: unknown device type 31
  <5>  Vendor: SONY      Model: USB-FDU           Rev: 5.01
  <5>  Type:   Unknown                            ANSI SCSI revision: 02
  <5>Attached scsi generic sg5 at scsi1, channel 0, id 0, lun 5,  type 31
  <6>scsi: unknown device type 31
  <5>  Vendor: SONY      Model: USB-FDU           Rev: 5.01
  <5>  Type:   Unknown                            ANSI SCSI revision: 02
  <5>Attached scsi generic sg6 at scsi1, channel 0, id 0, lun 6,  type 31
  <6>scsi: unknown device type 31
  <5>  Vendor: SONY      Model: USB-FDU           Rev: 5.01
  <5>  Type:   Unknown                            ANSI SCSI revision: 02
  <5>Attached scsi generic sg7 at scsi1, channel 0, id 0, lun 7,  type 31

                                                                 
T:  Bus=02 Lev=01 Prnt=01 Port=00 Cnt=01 Dev#=  5 Spd=12  MxCh= 0
D:  Ver= 1.10 Cls=00(>ifc ) Sub=00 Prot=00 MxPS= 8 #Cfgs=  1                  
P:  Vendor=054c ProdID=002c Rev= 5.01                                       
S:  Manufacturer=Sony                                                       
S:  Product=USB Floppy Drive                                               
C:* #Ifs= 1 Cfg#= 1 Atr=80 MxPwr=500mA                           
I:  If#= 0 Alt= 0 #EPs= 3 Cls=08(stor.) Sub=04 Prot=00 Driver=usb-storage   
E:  Ad=81(I) Atr=02(Bulk) MxPS=  64 Ivl=0ms                              
E:  Ad=02(O) Atr=02(Bulk) MxPS=  64 Ivl=0ms                              
E:  Ad=83(I) Atr=03(Int.) MxPS=   2 Ivl=16ms                             
                                                                                            

diff -purNx tags ../linux-2.6.11-rc3.orig/drivers/usb/storage/unusual_devs.h ./drivers/usb/storage/unusual_devs.h
--- ../linux-2.6.11-rc3.orig/drivers/usb/storage/unusual_devs.h	2005-02-03 02:56:33.000000000 +0100
+++ ./drivers/usb/storage/unusual_devs.h	2005-02-07 09:48:07.121999860 +0100
@@ -346,6 +346,13 @@ UNUSUAL_DEV(  0x054c, 0x002b, 0x0100, 0x
 		0 ),
 #endif
 
+/* Submitted by Olaf Hering, <olh@suse.de> SuSE Bugzilla #49049 */
+UNUSUAL_DEV(  0x054c, 0x002c, 0x0501, 0x0501,
+		"Sony",
+		"USB Floppy Drive",
+		US_SC_DEVICE, US_PR_DEVICE, NULL,
+		US_FL_SINGLE_LUN ),
+
 UNUSUAL_DEV(  0x054c, 0x002d, 0x0100, 0x0100, 
 		"Sony",
 		"Memorystick MSAC-US1",

